### PR TITLE
chore: switch workflows to use node 20 and 22 by default

### PR DESCRIPTION
**Pull request by marikaner**
Node 22 is the current node version, so all non-matrix workflows should run on this version by default.
Github actions seems to only allow `node16` and `node20` according to issues and [docs](https://docs.github.com/en/actions/sharing-automations/creating-actions/metadata-syntax-for-github-actions#runsusing-for-javascript-actions) that I found. I didn't try 22 though.

The tests are now running on all three versions.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        node-version: [18]
+        node-version: [20]
```
Comment (marikaner) [inline .github/workflows/build.yml:28]:
```suggestion
        node-version: [18, 20]
```

File: .github/workflows/blackduck.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
       - run: git fetch --depth=1
       - uses: actions/setup-node@v4
         with:
-          node-version: 18
+          node-version: 22
           cache: 'yarn'
       - run: yarn install --frozen-lockfile
```
Comment (ZhongpinWang) [inline .github/workflows/blackduck.yml:19]: `--ignore-engines` here as well?

Comment (marikaner) [inline .github/workflows/blackduck.yml:19]:
```suggestion
      - run: yarn install --frozen-lockfile --ignore-engines
```

Comment (marikaner) [reply .github/workflows/blackduck.yml:19]: yes! good catch

**Review by ZhongpinWang — APPROVED** [review]

**Pull request merged**